pkg/adaptor/cloud/azure/types.go

```diff
@@ -25,6 +27,37 @@ func (i *instanceSizes) Set(value string) error {
 	return nil
 }
 
+// keyValueFlag represents a flag of key-value pairs
+type keyValueFlag map[string]string
+
+// String returns the string representation of the keyValueFlag
+func (k *keyValueFlag) String() string {
+	var pairs []string
+	for key, value := range *k {
+		pairs = append(pairs, fmt.Sprintf("%s=%s", key, value))
+	}
+	return strings.Join(pairs, ", ")
+}
+
+// Set parses the input string and sets the keyValueFlag value
+func (k *keyValueFlag) Set(value string) error {
+	// Check if keyValueFlag is initialized. If not initialize it
+	if *k == nil {
+		*k = make(keyValueFlag, 0)
+	}
+	pairs := strings.Split(value, ",")
+	for _, pair := range pairs {
+		keyValue := strings.SplitN(pair, "=", 2)
+		if len(keyValue) != 2 {
```

Comment:
So empty values aren't supported, unlike k8s?

Reply:
Does the cloud provider (AWS/Azure) support empty values ? I need to re-check